lib/foreman/renderer.rb

```diff
@@ -57,6 +59,11 @@ def foreman_url(action = "built")
               :token     => (@host.token.value unless @host.try(:token).nil?), :kind => action
     end
 
+    def foreman_server_fqdn
+      config = URI.parse(Setting[:unattended_url])
+      config.host || request.host
```

Comment:
It sounds incorrect to force HTTPS, but what you're actually describing is that you prefer the value of :foreman_url setting and not the :unattended_url.